packages/terra-flex-box/src/FlexBox.jsx

```diff
@@ -0,0 +1,155 @@
+   */
+  alignContent: PropTypes.oneOf(alignContentTypes),
+  /**
+   * Defines the default behavior for how flex items are laid out along the cross axis on the current line.
+   */
+  alignItems: PropTypes.oneOf(alignTypes),
+  /**
+   * Allows the default alignment to be overridden for individual flex items.
+   */
+  alignSelf: PropTypes.oneOf(alignTypes),
+  /**
+   * Child nodes to be displayed in the container.
+   */
+  children: PropTypes.node,
+  /**
+   * Enables flex context for all its direct children either inline or block.
+   */
+  display: PropTypes.oneOf(['flex', 'inline-flex']),
+  /**
+   * Indicates what type of element the flex container can be.
+   */
+  element: PropTypes.oneOf(elementTypes),
+  /**
+   * Shorthand for flex-grow, flex-shrink and flex-basis combined.
+   */
+  flex: PropTypes.string,
+  /**
+   * Defines the default size of an element before the remaining space is distributed.
+   */
+  flexBasis: PropTypes.string,
+  /**
+   * Defines the direction flex items are placed in the container.
+   */
+  flexDirection: PropTypes.oneOf(['column', 'column-reverse', 'row', 'row-reverse']),
+  /**
+   * This is a shorthand flex-direction and flex-wrap properties, which together define the flex container's main and cross axes.
+   */
+  flexFlow: PropTypes.string,
+  /**
+   * Defines the ability for a flex item to grow if necessary.
+   */
+  flexGrow: PropTypes.number,
+  /**
+   * Defines the ability for a flex item to shrink if necessary.
+   */
+  flexShrink: PropTypes.number,
+  /**
+   * Allow the items to wrap as needed.
+   */
+  flexWrap: PropTypes.oneOf(['nowrap', 'wrap-reverse', 'wrap']),
+  /**
+   * Defines the height of the container.
+   */
+  height: PropTypes.string,
+  /**
+   * Defines the alignment along the main axis.
+   */
+  justifyContent: PropTypes.oneOf(justifyContentTypes),
+  /**
+   * Defines the maximum height for the container.
+   */
+  maxHeight: PropTypes.string,
+  /**
+   * Defines the maximum width for the container.
+   */
+  maxWidth: PropTypes.string,
+  /**
+   * Defines the minimum height for the container.
+   */
+  minHeight: PropTypes.string,
+  /**
+   * Defines the minimum width for the container.
+   */
+  minWidth: PropTypes.string,
+  /**
+   * Controls the order in which items appear in the container.
+   */
+  order: PropTypes.number,
+  /**
+   * Defines the width for the container.
+   */
+  width: PropTypes.string,
+};
+
+const defaultProps = {
+  display: 'flex',
+  element: 'div',
+};
+
+const FlexBox = ({
+  alignContent,
+  alignItems,
+  alignSelf,
+  children,
+  display,
+  element,
+  flex,
+  flexBasis,
+  flexDirection,
+  flexFlow,
+  flexGrow,
+  flexShrink,
+  flexWrap,
+  height,
+  justifyContent,
+  maxHeight,
+  maxWidth,
+  minHeight,
+  minWidth,
+  order,
+  width,
+  ...props
+}) => {
+  const customProps = props;
+  if (!customProps.style) { customProps.style = {}; }
```

Comment:
How are vendor prefixes and different flexbox syntaxes for IE handles with this? I don't believe this will work in IE unless we account for that.

Reply:
This goes back to my original question of what is the value proposition for this component. It seems to provide no simplification or advantage to just using flexbox html/css directly

Reply:
This goes back to my original question of what is the value proposition for this component. It seems to provide no simplification or advantage to just using flexbox html/css directly

Reply:
This goes back to my original question of what is the value proposition for this component. It seems to provide no simplification or advantage to just using flexbox html/css directly

Reply:
I see the value proposition as reducing duplicative styles. This component would reduce CSS needed every time we need CSS for flex properties with known values like display, align-content, align-items. We could write that CSS once and not need to repeat it. For properties which do not have a known set of values and can contain any value like width and height, theres no real trival way to prevent duplication there unless we provide a scale of "approved" values, but that seems limiting.

I do have concerns that this API may actually be too simple. My concern comes from working with flexbox and coming across issues where, to resolve a flexbox bug, you need to add additional styles, maybe not necessarily flexbox related. The [flexbug repo](https://github.com/philipwalton/flexbugs) has a list of various bugs where having the ability to add additional styles beyond flexbox based ones is required to achieve cross-browser functionality.

So I think in addition to providing vendor prefixes and different flexbox syntaxes for IE handled, we'd want this component to abstract common [flexbox bugs](https://github.com/philipwalton/flexbugs) as an implementation detail.

Whats the use case for this @mjhenkes @kschuste? My thought was this was something @mjhenkes and @mhemesath wanted.

Reply:
I had originally wanted one but upon seeing the tech design I'd rather prefer we create a subset of purposed "grid" components that solve reusable scenarios we encounter with flexbox.

Reply:
I'm looking for some basic tools for building simple layouts without having to duplicate the flexbox css and it's gotchas. I also like the idea of non devs (maybe) being able to use this in Kaiju, but if the API is to complex that may not work out.

Reply:
>  without having to duplicate the flexbox css

But now we've just pushed the problem up where instead of duplicating HTML/CSS we'll now duplicate implementations of this component.

Also, if we have duplications I'd much rather see us identify those and build dedicated components to solve those use cases.